robot/impl/local_robot.go

```diff
@@ -1096,6 +1100,16 @@ func dialRobotClient(
 // a best effort to remove no longer in use parts, but if it fails to do so, they could
 // possibly leak resources. The given config may be modified by Reconfigure.
 func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config) {
+	r.reconfigure(ctx, newConfig, false)
+}
+
+// ReconfigureSync is like Reconfigure, but prevents any resources from being processed
+// concurrently.
+func (r *localRobot) ReconfigureSync(ctx context.Context, newConfig *config.Config) {
```

Comment:
this is just intended for testing?

Reply:
at this point yes

Reply:
will either remove this or add a specific comment depending on how I change the parity test (per [your comment](https://github.com/viamrobotics/rdk/pull/3707#discussion_r1605277494))

Reply:
I ended up removing this test along with all test specific methods: https://github.com/viamrobotics/rdk/pull/3707/commits/75045ae7558530ada28639ac111f1d518d5d095d